frontend/server/src/Controllers/Tag.php

```diff
@@ -69,7 +69,25 @@ public static function getPublicTags() {
     /**
      * Return most frequent public tags of a certain level
      *
-     * @return list<array{alias: string}>
+     * @return list<array{alias: string, total: int}>
+     */
+    public static function getPublicQualityTagsByLevel(
+        string $problemLevel
+    ): array {
+        return \OmegaUp\Cache::getFromCacheOrSet(
+            \OmegaUp\Cache::TAGS_LIST,
+            "level-{$problemLevel}",
```

Comment:
nit:

```suggestion
            "publicquality-level-{$problemLevel}",
```

hay que intentar que la llave en el cache se parezca al nombre del método para evitar colisiones